plugins/inputs/apcupsd/apcupsd.go

```diff
@@ -0,0 +1,103 @@
+package apcupsd
+
+import (
+	"context"
+	"net/url"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/mdlayher/apcupsd"
+)
+
+const defaultAddress = "tcp://127.0.0.1:3551"
+
+var defaultTimeout = internal.Duration{Duration: time.Duration(time.Second * 5)}
+
+type ApcUpsd struct {
+	Servers []string
+	Timeout internal.Duration
+}
+
+func (_ *ApcUpsd) Description() string {
+	return "Monitor APC UPSes connected to apcupsd"
+}
+
+var sampleConfig = `
+  # a list of running apcupsd server to connect to. 
+  # If not provided will default to tcp://127.0.0.1:3551
+  servers = ["tcp://127.0.0.1:3551"]
+  timeout = "5s"
+`
+
+func (_ *ApcUpsd) SampleConfig() string {
+	return sampleConfig
+}
+
+func (h *ApcUpsd) Gather(acc telegraf.Accumulator) error {
+	for _, addr := range h.Servers {
+		addrBits, err := url.Parse(addr)
+		if err != nil {
+			return err
+		}
+		if addrBits.Scheme == "" {
+			addrBits.Scheme = "tcp"
+		}
+
+		status, err := fetchStatus(addrBits, h.Timeout.Duration)
+		if err != nil {
+			return err
+		}
+
+		tags := map[string]string{
+			"serial":   status.SerialNumber,
+			"ups_name": status.UPSName,
+			"online":   status.Status,
+		}
+
+		fields := map[string]interface{}{
+			"online":                 boolToInt(status.Status == "ONLINE"),
+			"input_voltage":          status.LineVoltage,
+			"load_percent":           status.LoadPercent,
+			"battery_charge_percent": status.BatteryChargePercent,
+			"time_left_minutes":      status.TimeLeft.Minutes(),
+			"output_voltage":         status.OutputVoltage,
+			"internal_temp":          status.InternalTemp,
+			"battery_voltage":        status.BatteryVoltage,
+			"input_frequency":        status.LineFrequency,
+			"time_on_battery":        status.TimeOnBattery.Minutes(),
```

Comment:
Add units: `time_on_battery_minutes` or `minutes_on_battery`

Reply:
I now believe here it would be best to convert to nanoseconds and report the field as `time_on_battery_ns`.  We are moving to this encoding for time durations and it can be shown easily in Grafana using the time -> nanoseconds unit.